examples/spiders/recaptcha.py

```diff
@@ -0,0 +1,51 @@
+import scrapy
+from twisted.python.failure import Failure
+
+from scrapypuppeteer import PuppeteerRequest
+from scrapypuppeteer.actions import GoTo, RecaptchaSolver, Click, Screenshot
+from scrapypuppeteer.response import PuppeteerResponse, PuppeteerJsonResponse
+
+
+class RecaptchaSpider(scrapy.Spider):
+    name = "recaptcha"
+
+    start_urls = ["https://www.google.com/recaptcha/api2/demo"]
+
+    def start_requests(self):
+        for url in self.start_urls:
+            action = GoTo(url=url)
+            yield PuppeteerRequest(action=action, callback=self.parse_html, errback=self.error, close_page=False)
+
+    def solve_recaptcha(self, response: PuppeteerResponse, **kwargs):
```

Comment:
Not used. I believe you have used this spider both for explicit and middleware-based captcha testing. Make two test spiders instead, demonstrating both approaches.